### ISSUE TITLE
Dubte normalització

### ISSUE THREAD

**Issue by victor-prt** [gdsa-upc/gdsa-suport#20]
Hola, volía preguntar si a l'hora de fer l'entrenament del codebook hem de normalitzar els descriptors de la imatge i en el cas de que sigui així si podem retornar els descriptors normalitzats ja des de la funcio get_local_features.

**Comment by amaiasalvador** [comment]
Hola Víctor,

no és necessari normalitzar els descriptors abans d'entrenar el codebook, però això no vol dir que estigui prohibit. És possible que els resultats millorin si normalitzeu , o que justament passi el contrari. Podeu provar les dues coses. Recordeu però que qualsevol normalització o transformació que apliqueu als descriptors per entrenar el codebook cal aplicar-la també en el moment de calcular els assignments.

En qualsevol cas, si voleu normalitzar amb norma [L2 o L1](http://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.normalize.html#sklearn.preprocessing.normalize), ho podeu fer a dins de la funció _get_local_features_ directament, ja que la normalització s'aplica independentment per cada descriptor local.

**Comment by xavigiro** [comment]
La proposta que fa l'Amaia d'experimentar és molt interessant. Per ara, però, el meu consell és que seguiu la proposta estàndard i normalitzeu la signatura (histograma) que resulta d'assignar els descriptors extrets d'una imatge a les paraules del vocabulari.

A nivell de normalitzar el descriptor, hi ha [autors](https://www.robots.ox.ac.uk/~vgg/publications/2012/Arandjelovic12/presentation.pdf) que diuen que treballar amb el [RootSIFT](http://www.pyimagesearch.com/2015/04/13/implementing-rootsift-in-python-and-opencv/) és millor que amb el SIFT.
